### PR TITLE
fix the case when user must not have any supplementary groups

### PR DESCRIPTION
**Pull request by arachnopavel**
Fix the case when user must not have any supplementary groups, by supplying the -G option with empty group list.

### CODE REVIEW

**Comment by AlexanderS** [comment]
Seems like the test needs to be adjusted.

File: src/lib/Bcfg2/Client/Tools/POSIXUsers.py [hunk 1]
```diff
@@ -261,6 +261,8 @@ def _get_cmd(self, action, entry):
                           for e in entry.findall("MemberOf")]
                 if extras:
                     cmd.extend(['-G', ",".join(extras)])
+                else:
+                    cmd.extend(['-G', ''])
```
Comment (solj) [inline src/lib/Bcfg2/Client/Tools/POSIXUsers.py]: Could we achieve this same effect by doing something like https://gist.github.com/solj/a1d89b429e895471f275c1387f67418e? This seems like a simpler route.

Comment (arachnopavel) [inline src/lib/Bcfg2/Client/Tools/POSIXUsers.py]: I'd prefer my way because it's more explicit, just in case.

Comment (arachnopavel) [inline src/lib/Bcfg2/Client/Tools/POSIXUsers.py]: Is it going to be accepted, or should I refactor this particular change first?

**Comment by arachnopavel** [comment]
Are there any issues with these commits that need to be resolved on my part?

**Comment by AlexanderS** [comment]
FYI: I just cherry picked the XMLSyntaxError commit to master c5c90ee87a93b778117d24bf822152eb7582dca3

**Comment by solj** [comment]
:+1:

**Comment by AlexanderS** [comment]
Merged, thanks!

**Pull request merged**